src/server/handlers/github.js

```diff
@@ -99,13 +100,18 @@ export const getSnapName = (repositoryUrl, token) => {
 
       return internalGetSnapcraftYaml(owner, name, token)
         .then((snapcraftYaml) => {
-          const snapName = snapcraftYaml.name;
-          if (snapName) {
-            getMemcached().set(cacheId, snapName, 3600);
-            return resolve(snapName);
-          } else {
-            return resolve(null);
+          const snapcraftData = {};
+          for (let index in snapcraftYaml) {
+            if (snapcraftYaml.hasOwnProperty(index)) {
+              if (SNAPCRAFT_INFO_WHITELIST.indexOf(index) >= 0) {
+                snapcraftData[index] = snapcraftYaml[index];
+              }
+            }
           }
```

Comment:
I've replaced storage of the snap name in Memcached with an object of whitelisted items from snapcraft.yml. This felt like the best way to cache both the snap name and the existence of the snapcraft.yml file.